From 9016b0a6b31b5a24367c8574c60c84956aa0f36a Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Thu, 21 Oct 2021 18:59:25 -0700
Subject: [PATCH 4/4] Add Android 12 workaround for PRODUCT_COPY_FILES

---
 scripts/generate-vendor.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/scripts/generate-vendor.sh b/scripts/generate-vendor.sh
index 6b0c4be..e38c204 100755
--- a/scripts/generate-vendor.sh
+++ b/scripts/generate-vendor.sh
@@ -501,6 +501,9 @@ gen_board_cfg_mk() {
     # Android 11 workaround for including .vintf files in PRODUCT_COPY_FILES
     echo 'BUILD_BROKEN_VINTF_PRODUCT_COPY_FILES := true'
 
+    # Android 12 workaround. We should be using cc_prebuilt_binary / cc_prebuilt_library_shared
+    echo 'BUILD_BROKEN_ELF_PREBUILT_PRODUCT_COPY_FILES := true'
+
     # Update with user selected extra flags
     echo "$MK_FLAGS_LIST"
   } >> "$BOARD_CONFIG_VENDOR_MK"
-- 
2.33.0

