diff --git a/aosp_blueline.mk b/aosp_blueline.mk
index 9a38f9fb..4f657ae0 100644
--- a/aosp_blueline.mk
+++ b/aosp_blueline.mk
@@ -30,4 +30,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_BRAND := Android
 PRODUCT_NAME := aosp_blueline
 PRODUCT_DEVICE := blueline
-PRODUCT_MODEL := AOSP on blueline
+PRODUCT_MODEL := Pixel 3
diff --git a/aosp_crosshatch.mk b/aosp_crosshatch.mk
index 7db148cf..a54202c4 100644
--- a/aosp_crosshatch.mk
+++ b/aosp_crosshatch.mk
@@ -30,4 +30,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_BRAND := Android
 PRODUCT_NAME := aosp_crosshatch
 PRODUCT_DEVICE := crosshatch
-PRODUCT_MODEL := AOSP on crosshatch
+PRODUCT_MODEL := Pixel 3 XL
