diff --git a/crosshatch/AndroidBoard.mk b/crosshatch/AndroidBoard.mk
index 0099fda6..660fec6c 100644
--- a/crosshatch/AndroidBoard.mk
+++ b/crosshatch/AndroidBoard.mk
@@ -25,7 +25,7 @@ INSTALLED_PERSISTIMAGE_TARGET := $(PRODUCT_OUT)/persist.img
 $(INSTALLED_PERSISTIMAGE_TARGET): $(MKEXTUSERIMG) $(MAKE_EXT4FS) $(INSTALLED_PERSISTIMAGE_FILES)
 	$(call pretty,"Target persist fs image: $(INSTALLED_PERSISTIMAGE_TARGET)")
 	@mkdir -p $(TARGET_OUT_PERSIST_IMG_PATH)
-	$(hide) PATH=$(HOST_OUT_EXECUTABLES):$${PATH} $(MKEXTUSERIMG) -s $(TARGET_OUT_PERSIST_IMG_PATH) $@ ext4 persist $(BOARD_PERSISTIMAGE_PARTITION_SIZE)
+	$(hide) PATH=$(HOST_OUT_EXECUTABLES):$${PATH} $(MKEXTUSERIMG) -U @uuid@ -S @hashSeed@ -T @buildDateTime@ -s $(TARGET_OUT_PERSIST_IMG_PATH) $@ ext4 persist $(BOARD_PERSISTIMAGE_PARTITION_SIZE)
 	$(hide) chmod a+r $@
 	$(hide) $(call assert-max-image-size,$@,$(BOARD_PERSISTIMAGE_PARTITION_SIZE))
 
