diff --git a/app/src/main/java/org/fdroid/fdroid/installer/ApkVerifier.java b/app/src/main/java/org/fdroid/fdroid/installer/ApkVerifier.java
index fc827122a..c2ebd2b08 100644
--- a/app/src/main/java/org/fdroid/fdroid/installer/ApkVerifier.java
+++ b/app/src/main/java/org/fdroid/fdroid/installer/ApkVerifier.java
@@ -114,11 +114,9 @@ class ApkVerifier {
         if (expected == null || actual == null) {
             return false;
         }
-        if (expected.length != actual.length) {
-            return false;
-        }
         HashSet<String> expectedSet = new HashSet<>(Arrays.asList(expected));
         HashSet<String> actualSet = new HashSet<>(Arrays.asList(actual));
+        actualSet.remove("android.permission.OTHER_SENSORS"); // Ignore GrapheneOS compat added permission that wouldn't be in the listed permissions anyway
         return expectedSet.equals(actualSet);
     }
 
