diff --git a/pkgs/development/mobile/androidenv/compose-android-packages.nix b/pkgs/development/mobile/androidenv/compose-android-packages.nix
index 935a4a4f2fb..ecea7705d74 100644
--- a/pkgs/development/mobile/androidenv/compose-android-packages.nix
+++ b/pkgs/development/mobile/androidenv/compose-android-packages.nix
@@ -238,6 +238,14 @@ rec {
           ln -s ${addon}/libexec/android-sdk/${path} $targetDir
         '') includeExtras}
 
+      mkdir  -p licenses
+      echo 601085b94cd77f0b54ff86406957099ebe79c4d6 >> licenses/android-googletv-license
+      echo d56f5187479451eabf01fb78af6dfcb131a6481e >> licenses/android-sdk-license
+      echo 24333f8a63b6825ea9c5514f83c2829b004d1fee >> licenses/android-sdk-license
+      echo 84831b9409646a918e30573bab4c9c91346d8abd >> licenses/android-sdk-preview-license
+      echo 33b6a2b64607f11b759f320ef9dff4ae5c47d97a >> licenses/google-sdk-license
+      echo e9acab5b5fbb560a72cfaecce8946896ff6aab9d >> licenses/mips-android-sysimage-license
+
       # Expose common executables in bin/
       mkdir -p $out/bin
       find $PWD/tools -not -path '*/\.*' -type f -executable -mindepth 1 -maxdepth 1 | while read i
