From 5af943e5b9378220db65d5a2acc1f4b6f6fe409d Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Thu, 16 Apr 2020 14:15:52 -0400
Subject: [PATCH 12/16] dumpjson: schema change to use relpath as keys

---
 subcmds/dumpjson.py | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/subcmds/dumpjson.py b/subcmds/dumpjson.py
index 176391c..2105002 100644
--- a/subcmds/dumpjson.py
+++ b/subcmds/dumpjson.py
@@ -91,18 +91,17 @@ class Dumpjson(Command, MirrorSafeCommand):
 
     data = {}
     for p in all_projects:
-        data[p.name] = {
+        data[p.relpath] = {
             "url": p.remote.url,
-            "relpath": p.relpath,
             "revisionExpr": p.revisionExpr,
             "rev": p.rev,
         }
         filtered_groups = filter(lambda g: not (g == "all" or g.startswith("name:") or g.startswith("path:")), p.groups)
         if filtered_groups:
-            data[p.name]["groups"] = sorted(filtered_groups)
+            data[p.relpath]["groups"] = sorted(filtered_groups)
         if p.linkfiles:
-            data[p.name]["linkfiles"] = [ { "src": l.src, "dest": l.dest } for l in p.linkfiles ]
+            data[p.relpath]["linkfiles"] = [ { "src": l.src, "dest": l.dest } for l in p.linkfiles ]
         if p.copyfiles:
-            data[p.name]["copyfiles"] = [ { "src": c.src, "dest": c.dest } for c in p.copyfiles ]
+            data[p.relpath]["copyfiles"] = [ { "src": c.src, "dest": c.dest } for c in p.copyfiles ]
 
     print(json.dumps(data, sort_keys=True))
-- 
2.44.0

