diff --git a/aosp_bonito.mk b/aosp_bonito.mk
index c1768593..1d06ab07 100644
--- a/aosp_bonito.mk
+++ b/aosp_bonito.mk
@@ -57,4 +57,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_BRAND := Android
 PRODUCT_NAME := aosp_bonito
 PRODUCT_DEVICE := bonito
-PRODUCT_MODEL := AOSP on bonito
+PRODUCT_MODEL := Pixel 3a XL
diff --git a/aosp_sargo.mk b/aosp_sargo.mk
index e7c148e2..e38b7be6 100644
--- a/aosp_sargo.mk
+++ b/aosp_sargo.mk
@@ -57,4 +57,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_BRAND := Android
 PRODUCT_NAME := aosp_sargo
 PRODUCT_DEVICE := sargo
-PRODUCT_MODEL := AOSP on sargo
+PRODUCT_MODEL := Pixel 3a
