diff --git a/aosp_coral.mk b/aosp_coral.mk
index 27c6967..a2b0583 100644
--- a/aosp_coral.mk
+++ b/aosp_coral.mk
@@ -60,4 +60,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_BRAND := Android
 PRODUCT_NAME := aosp_coral
 PRODUCT_DEVICE := coral
-PRODUCT_MODEL := AOSP on coral
+PRODUCT_MODEL := Pixel 4 XL
diff --git a/aosp_flame.mk b/aosp_flame.mk
index 47ac10f..8cf61dd 100644
--- a/aosp_flame.mk
+++ b/aosp_flame.mk
@@ -61,4 +61,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_BRAND := Android
 PRODUCT_NAME := aosp_flame
 PRODUCT_DEVICE := flame
-PRODUCT_MODEL := AOSP on flame
+PRODUCT_MODEL := Pixel 4
