diff --git a/aosp_blueline.mk b/aosp_blueline.mk
index 624d5da6..d048b43a 100644
--- a/aosp_blueline.mk
+++ b/aosp_blueline.mk
@@ -56,4 +56,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_NAME := aosp_blueline
 PRODUCT_DEVICE := blueline
 PRODUCT_BRAND := Android
-PRODUCT_MODEL := AOSP on blueline
+PRODUCT_MODEL := Pixel 3
diff --git a/aosp_crosshatch.mk b/aosp_crosshatch.mk
index c921d366..6c5efed1 100644
--- a/aosp_crosshatch.mk
+++ b/aosp_crosshatch.mk
@@ -56,4 +56,4 @@ PRODUCT_MANUFACTURER := Google
 PRODUCT_NAME := aosp_crosshatch
 PRODUCT_DEVICE := crosshatch
 PRODUCT_BRAND := Android
-PRODUCT_MODEL := AOSP on crosshatch
+PRODUCT_MODEL := Pixel 3 XL
