diff --git a/aosp_marlin.mk b/aosp_marlin.mk
index 96cc5e95..1b9d9e7f 100644
--- a/aosp_marlin.mk
+++ b/aosp_marlin.mk
@@ -24,7 +24,7 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_base_telephony.mk)
 PRODUCT_NAME := aosp_marlin
 PRODUCT_DEVICE := marlin
 PRODUCT_BRAND := Android
-PRODUCT_MODEL := AOSP on msm8996
+PRODUCT_MODEL := Pixel XL
 PRODUCT_MANUFACTURER := Google
 PRODUCT_RESTRICT_VENDOR_FILES := true
 
diff --git a/aosp_sailfish.mk b/aosp_sailfish.mk
index b6e6427c..675f11a6 100644
--- a/aosp_sailfish.mk
+++ b/aosp_sailfish.mk
@@ -24,7 +24,7 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/aosp_base_telephony.mk)
 PRODUCT_NAME := aosp_sailfish
 PRODUCT_DEVICE := sailfish
 PRODUCT_BRAND := Android
-PRODUCT_MODEL := AOSP on msm8996
+PRODUCT_MODEL := Pixel
 PRODUCT_MANUFACTURER := Google
 PRODUCT_RESTRICT_VENDOR_FILES := true
