diff -Naur a/aosp_marlin.mk b/aosp_marlin.mk
--- a/aosp_marlin.mk	2019-06-12 15:06:33.434485524 -0400
+++ b/aosp_marlin.mk	2019-06-12 15:10:19.511867098 -0400
@@ -24,8 +24,8 @@
 PRODUCT_NAME := aosp_marlin
 PRODUCT_DEVICE := marlin
 PRODUCT_BRAND := Android
-PRODUCT_MODEL := AOSP on msm8996
-PRODUCT_MANUFACTURER := google
+PRODUCT_MODEL := Pixel XL
+PRODUCT_MANUFACTURER := Google
 PRODUCT_RESTRICT_VENDOR_FILES := true
 
 PRODUCT_COPY_FILES += device/google/marlin/fstab.common:$(TARGET_COPY_OUT_VENDOR)/etc/fstab.marlin
diff -Naur a/aosp_sailfish.mk b/aosp_sailfish.mk
--- a/aosp_sailfish.mk	2019-06-12 15:06:33.429485449 -0400
+++ b/aosp_sailfish.mk	2019-06-12 15:10:37.744140466 -0400
@@ -24,8 +24,8 @@
 PRODUCT_NAME := aosp_sailfish
 PRODUCT_DEVICE := sailfish
 PRODUCT_BRAND := Android
-PRODUCT_MODEL := AOSP on msm8996
-PRODUCT_MANUFACTURER := google
+PRODUCT_MODEL := Pixel
+PRODUCT_MANUFACTURER := Google
 PRODUCT_RESTRICT_VENDOR_FILES := true
 
 PRODUCT_COPY_FILES += device/google/marlin/fstab.common:$(TARGET_COPY_OUT_VENDOR)/etc/fstab.sailfish
