diff --git a/tools/releasetools/build_image.py b/tools/releasetools/build_image.py
index 4136ed432..23fa83508 100755
--- a/tools/releasetools/build_image.py
+++ b/tools/releasetools/build_image.py
@@ -57,8 +57,8 @@ def GetDiskUsage(path):
   Returns:
     The number of bytes based on a 1K block_size.
   """
-  cmd = ["du", "-k", "-s", path]
-  output = common.RunAndCheckOutput(cmd, verbose=False)
+  cmd = ["@coreutils@/bin/du", "--apparent-size", "-k", "-s", path]
+  output = common.RunAndCheckOutput(cmd, verbose=True)
   return int(output.split()[0]) * 1024
 
 
