From 397bf51627d43c483bdfbb7c57911c312a6cea92 Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Fri, 1 Oct 2021 19:15:34 -0700
Subject: [PATCH 4/4] Add option to use externally provided carrier_list.pb

---
 execute-all.sh | 33 +++++++++++++++++++++------------
 1 file changed, 21 insertions(+), 12 deletions(-)

diff --git a/execute-all.sh b/execute-all.sh
index 66399e4..6c265a3 100755
--- a/execute-all.sh
+++ b/execute-all.sh
@@ -62,12 +62,13 @@ usage() {
 cat <<_EOF
   Usage: $(basename "$0") [options]
     OPTIONS:
-      -d|--device <name> : Device codename (angler, bullhead, etc.)
-      -a|--alias <alias> : Device alias (e.g. flounder volantis (WiFi) vs volantisg (LTE))
-      -b|--buildID <id>  : BuildID string (e.g. MMB29P)
-      -o|--output <path> : Path to save generated vendor data
-      -i|--img <path>    : [OPTIONAL] Read factory image archive from file instead of downloading
-      -O|--ota <path>    : [OPTIONAL] Read OTA image archive from file instead of downloading
+      -d|--device <name>              : Device codename (angler, bullhead, etc.)
+      -a|--alias <alias>              : Device alias (e.g. flounder volantis (WiFi) vs volantisg (LTE))
+      -b|--buildID <id>               : BuildID string (e.g. MMB29P)
+      -o|--output <path>              : Path to save generated vendor data
+      -i|--img <path>                 : [OPTIONAL] Read factory image archive from file instead of downloading
+      -O|--ota <path>                 : [OPTIONAL] Read OTA image archive from file instead of downloading
+      -C|--carrier-list-folder <path> : [OPTIONAL] Path to folder containing carrier_list.pb instead of downloading
       -j|--java <path    : [OPTIONAL] Java path to use instead of system auto detected global version
       -f|--full    : [OPTIONAL] Use config with all non-essential OEM blobs to be compatible with GApps (default: false)
       -k|--keep    : [OPTIONAL] Keep all extracted factory images & repaired data (default: false)
@@ -389,6 +390,10 @@ do
       INPUT_OTA="$(_realpath "$2")"
       shift
       ;;
+    -C|--carrier-list-folder)
+      CARRIER_LIST_FOLDER="$(_realpath "$2")"
+      shift
+      ;;
     -f|--full)
       CONFIG_TYPE="full"
       ;;
@@ -569,12 +574,16 @@ if [[ "$OtaArchive" == "" ]]; then
 fi
 fi
 
-# Download carrier list
-aospCarrierListFolder="$SCRIPTS_ROOT/scripts/carriersettings-extractor"
-$DOWNLOAD_CARRIER_LIST_SCRIPT --output "$aospCarrierListFolder" || {
-  echo "[-] Carrier list download failed"
-  abort 1
-}
+# Download carrier list if not provided
+if [[ "$CARRIER_LIST_FOLDER" == "" ]]; then
+  aospCarrierListFolder="$SCRIPTS_ROOT/scripts/carriersettings-extractor"
+  $DOWNLOAD_CARRIER_LIST_SCRIPT --output "$aospCarrierListFolder" || {
+    echo "[-] Carrier list download failed"
+    abort 1
+  }
+else
+  aospCarrierListFolder="$CARRIER_LIST_FOLDER"
+fi
 
 # Clear old data if present & extract data from factory images
 if [ -d "$FACTORY_IMGS_DATA" ]; then
-- 
2.32.0

