From 09e077d64256a4474de32b5ac793bad5be34fa72 Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Fri, 6 Sep 2019 16:59:03 -0400
Subject: [PATCH 1/4] Disable oatdump update

---
 execute-all.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/execute-all.sh b/execute-all.sh
index 62e21c1..25f28f0 100755
--- a/execute-all.sh
+++ b/execute-all.sh
@@ -176,10 +176,10 @@ oatdump_prepare_env() {
     oatdump_deps_download "$api_level"
   fi
 
-  if needs_oatdump_update "$api_level"; then
-    echo "[*] Outdated version detected - downloading oatdump host bin & lib dependencies"
-    oatdump_deps_download "$api_level"
-  fi
+#  if needs_oatdump_update "$api_level"; then
+#    echo "[*] Outdated version detected - downloading oatdump host bin & lib dependencies"
+#    oatdump_deps_download "$api_level"
+#  fi
 }
 
 is_aosp_root() {
-- 
2.22.0


From f8e3cc8d46bde0adb63d554b5c1c93eec39ede49 Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Fri, 6 Sep 2019 16:59:30 -0400
Subject: [PATCH 3/4] Just write proprietary-blobs.txt to current dir

---
 execute-all.sh             | 2 +-
 scripts/generate-vendor.sh | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/execute-all.sh b/execute-all.sh
index 25f28f0..124305a 100755
--- a/execute-all.sh
+++ b/execute-all.sh
@@ -567,7 +567,7 @@ echo "[*] Processing with 'API-$API_LEVEL $CONFIG_TYPE' configuration"
 
 # Generate unified readonly "proprietary-blobs.txt"
 $GEN_BLOBS_LIST_SCRIPT --input "$FACTORY_IMGS_DATA/vendor" \
-    --output "$SCRIPTS_ROOT/$DEVICE" \
+    --output . \
     --api "$API_LEVEL" \
     --conf-file "$CONFIG_FILE" \
     --conf-type "$CONFIG_TYPE" || {
diff --git a/scripts/generate-vendor.sh b/scripts/generate-vendor.sh
index d022597..0653cc7 100755
--- a/scripts/generate-vendor.sh
+++ b/scripts/generate-vendor.sh
@@ -1116,7 +1116,7 @@ isValidApiLevel "$API_LEVEL"
 
 # Populate config files from base conf dir
 readonly DEVICE_CONFIG_DIR="$(dirname "$CONFIG_FILE")"
-readonly BLOBS_LIST="$DEVICE_CONFIG_DIR/proprietary-blobs.txt"
+readonly BLOBS_LIST="./proprietary-blobs.txt"
 readonly OVERLAYS_DIR="$(setOverlaysDir)"
 readonly DEP_DSO_BLOBS_LIST="$(jqIncRawArray "$API_LEVEL" "$CONFIG_TYPE" "dep-dso" "$CONFIG_FILE" | grep -Ev '(^#|^$)')"
 readonly MK_FLAGS_LIST="$(jqIncRawArray "$API_LEVEL" "$CONFIG_TYPE" "BoardConfigVendor" "$CONFIG_FILE")"
-- 
2.22.0


From 5746cb09da6e45300501d3f6fc59434e93dada70 Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Wed, 11 Sep 2019 13:02:03 -0400
Subject: [PATCH 4/4] Allow for externally-set config file

---
 execute-all.sh             | 8 +++++++-
 scripts/generate-vendor.sh | 5 ++---
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/execute-all.sh b/execute-all.sh
index 124305a..bed91f1 100755
--- a/execute-all.sh
+++ b/execute-all.sh
@@ -415,6 +415,10 @@ do
       TIMESTAMP="$2"
       shift
       ;;
+    --conf-file)
+      CONFIG_FILE="$(_realpath $2)"
+      shift
+      ;;
     *)
       echo "[-] Invalid argument '$1'"
       usage
@@ -467,7 +471,9 @@ update_java_path
 check_supported_device
 
 # Check supported API for device
-CONFIG_FILE="$SCRIPTS_ROOT/$DEVICE/config.json"
+if [[ "$CONFIG_FILE" = "" ]]; then
+    CONFIG_FILE="$SCRIPTS_ROOT/$DEVICE/config.json"
+fi
 
 # Prepare output dir structure
 OUT_BASE="$OUTPUT_DIR/$DEVICE/$BUILDID"
diff --git a/scripts/generate-vendor.sh b/scripts/generate-vendor.sh
index 0653cc7..c1c7817 100755
--- a/scripts/generate-vendor.sh
+++ b/scripts/generate-vendor.sh
@@ -1115,7 +1115,8 @@ isValidConfigType "$CONFIG_TYPE"
 isValidApiLevel "$API_LEVEL"
 
 # Populate config files from base conf dir
-readonly DEVICE_CONFIG_DIR="$(dirname "$CONFIG_FILE")"
+DEVICE=$(get_device_codename "$INPUT_DIR/system/build.prop")
+readonly DEVICE_CONFIG_DIR="$SCRIPTS_DIR/../$DEVICE"
 readonly BLOBS_LIST="./proprietary-blobs.txt"
 readonly OVERLAYS_DIR="$(setOverlaysDir)"
 readonly DEP_DSO_BLOBS_LIST="$(jqIncRawArray "$API_LEVEL" "$CONFIG_TYPE" "dep-dso" "$CONFIG_FILE" | grep -Ev '(^#|^$)')"
@@ -1135,7 +1136,6 @@ readarray -t PSIG_BC_FILES < <(
 verify_input "$INPUT_DIR"
 
 # Get device details
-DEVICE=$(get_device_codename "$INPUT_DIR/system/build.prop")
 DEVICE_FAMILY="$(jqRawStrTop "device-family" "$CONFIG_FILE")"
 VENDOR=$(get_vendor "$INPUT_DIR/system/build.prop")
 VENDOR_DIR="$(jqRawStrTop "aosp-vendor-dir" "$CONFIG_FILE")"
@@ -1146,7 +1146,6 @@ if [[ "$EXTRA_IMGS_LIST" != "" ]]; then
   readarray -t EXTRA_IMGS < <(echo "$EXTRA_IMGS_LIST")
 fi
 
-
 echo "[*] Generating '$DEVICE' vendor blobs"
 
 # Prepare vendor output directory structure
-- 
2.22.0

