diff --git a/src/blobs/build.ts b/src/blobs/build.ts
index 825ef2af..c94ccd3e 100644
--- a/src/blobs/build.ts
+++ b/src/blobs/build.ts
@@ -212,14 +212,9 @@ function getVersionCheckFilePath(dirs: VendorDirectories) {
 }
 
 export async function writeVersionCheckFile(config: DeviceConfig, dirs: VendorDirectories) {
-  let adevtoolRevision = await spawnAsync('git', ['-C', ADEVTOOL_DIR, 'rev-parse', 'HEAD'])
-  assert(adevtoolRevision.endsWith('\n'))
-  adevtoolRevision = adevtoolRevision.slice(0, -1)
+  let adevtoolRevision = '@adevtoolRevision@'
   let deviceName = config.device.name
-  let contents =
-    `ifneq ($(shell git -C ${RELATIVE_ADEVTOOL_PATH} rev-parse HEAD),${adevtoolRevision})\n` +
-    `  $(error ${deviceName} vendor module is outdated. Run \`adevtool generate-all -d ${deviceName}\` to update it)\n` +
-    `endif`
+  let contents = ''
   await fs.writeFile(getVersionCheckFilePath(dirs), contents)
 }
 
diff --git a/src/config/paths.ts b/src/config/paths.ts
index 7deda719..fffd39b5 100644
--- a/src/config/paths.ts
+++ b/src/config/paths.ts
@@ -49,7 +49,7 @@ export async function getHostBinPath(programName: string) {
   let outDir = 'out_adevtool_deps'
 
   async function getAdevtoolRevision() {
-    return spawnAsync('git', ['-C', ADEVTOOL_DIR, 'rev-parse', 'HEAD'])
+    return `@adevtoolRevision@\n`
   }
 
   function getAdevtoolRevisionFile(outDir: string) {
