diff --git a/src/app/seamlessupdate/client/BootReceiver.java b/src/app/seamlessupdate/client/BootReceiver.java
index 3be6d86..c9b397f 100644
--- a/src/app/seamlessupdate/client/BootReceiver.java
+++ b/src/app/seamlessupdate/client/BootReceiver.java
@@ -14,10 +14,6 @@ public class BootReceiver extends BroadcastReceiver {
             final SharedPreferences preferences = Settings.getPreferences(context);
             preferences.edit().putBoolean(Settings.KEY_WAITING_FOR_REBOOT, false).apply();
             PeriodicJob.schedule(context);
-
-            if (!preferences.contains(Settings.KEY_USE_SECURITY_PREVIEW_CHANNEL)) {
-                NotificationHandler.showSetSecurityPreviewNotification(context);
-            }
         } else {
             context.getPackageManager().setApplicationEnabledSetting(context.getPackageName(),
                 PackageManager.COMPONENT_ENABLED_STATE_DISABLED, 0);
