diff --git a/app/src/main/java/org/fdroid/fdroid/privileged/ClientWhitelist.java b/app/src/main/java/org/fdroid/fdroid/privileged/ClientWhitelist.java
index 993fadb..fc88f72 100644
--- a/app/src/main/java/org/fdroid/fdroid/privileged/ClientWhitelist.java
+++ b/app/src/main/java/org/fdroid/fdroid/privileged/ClientWhitelist.java
@@ -34,8 +34,8 @@ import java.util.HashSet;
 public class ClientWhitelist {
 
     public static HashSet<Pair<String, String>> whitelist = new HashSet<>(Arrays.asList(
-            // certificate SHA-256 of https//f-droid.org/F-Droid.apk
-            new Pair<>("org.fdroid.fdroid", "43238d512c1e5eb2d6569f4a3afbf5523418b82e0a3ed1552770abb9a9c9ccab")
+            // Custom certificate
+            new Pair<>("org.fdroid.fdroid", "@fingerprint@")
     ));
 
 }
