diff --git a/common/libs/utils/network.cpp b/common/libs/utils/network.cpp
index df91c23e..eb47cc5c 100644
--- a/common/libs/utils/network.cpp
+++ b/common/libs/utils/network.cpp
@@ -136,7 +136,7 @@ SharedFD OpenTapInterface(const std::string& interface_name) {
 }
 
 std::set<std::string> TapInterfacesInUse() {
-  Command cmd("/bin/bash");
+  Command cmd("@bash@/bin/bash");
   cmd.AddParameter("-c");
   cmd.AddParameter("egrep -h -e \"^iff:.*\" /proc/*/fdinfo/*");
   std::string stdin, stdout, stderr;
diff --git a/host/libs/config/cuttlefish_config.cpp b/host/libs/config/cuttlefish_config.cpp
index ea5c4723..28c579f3 100644
--- a/host/libs/config/cuttlefish_config.cpp
+++ b/host/libs/config/cuttlefish_config.cpp
@@ -913,14 +913,14 @@ std::string DefaultGuestImagePath(const std::string& file_name) {
 bool HostSupportsQemuCli() {
   static bool supported =
       std::system(
-          "/usr/lib/cuttlefish-common/bin/capability_query.py qemu_cli") == 0;
+          "@cuttlefish_common@/bin/capability_query.py qemu_cli") == 0;
   return supported;
 }
 
 bool HostSupportsVsock() {
   static bool supported =
       std::system(
-          "/usr/lib/cuttlefish-common/bin/capability_query.py vsock") == 0;
+          "@cuttlefish_common@/bin/capability_query.py vsock") == 0;
   return supported;
 }
 }  // namespace cuttlefish
