diff --git a/avbtool.py b/avbtool.py
index aceb97fb..583ec29c 100755
--- a/avbtool.py
+++ b/avbtool.py
@@ -39,6 +39,8 @@ import sys
 import tempfile
 import time
 
+import stat
+
 # Keep in sync with libavb/avb_version.h.
 AVB_VERSION_MAJOR = 1
 AVB_VERSION_MINOR = 3
@@ -804,6 +806,7 @@ class ImageHandler(object):
     if self._read_only:
       self._image = open(self.filename, 'rb')
     else:
+      os.chmod(self.filename, stat.S_IRWXU | stat.S_IRWXG | stat.S_IROTH)
       self._image = open(self.filename, 'r+b')
     self._image.seek(0, os.SEEK_END)
     self.image_size = self._image.tell()
