From 4877076e4450d760b05928fdf6ca8fac96dab972 Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Fri, 29 May 2020 18:14:48 -0400
Subject: [PATCH 2/2] HACK: prefix tool names

---
 diffoscope/external_tools.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/diffoscope/external_tools.py b/diffoscope/external_tools.py
index e96ab71b..4c6cb703 100644
--- a/diffoscope/external_tools.py
+++ b/diffoscope/external_tools.py
@@ -223,7 +223,10 @@ EXTERNAL_TOOLS = {
 # May be populated at runtime by remapped names like
 # readelf -> arm-none-eabi-readelf, etc
 # diff -> gdiff, etc
-REMAPPED_TOOL_NAMES = {}
+REMAPPED_TOOL_NAMES = {
+    'readelf': 'aarch64-unknown-linux-gnu-readelf',
+    'objdump': 'aarch64-unknown-linux-gnu-objdump',
+}
 
 # GNU programs whose names differ on some non-GNU systems such as FreeBSD etc
 # AND where the CLI or output of the programs differ from the non-GNU system
-- 
2.28.0

