From 981df6e92ba00102be8b3dc49b876f94ed726529 Mon Sep 17 00:00:00 2001
From: Daniel Fullmer <danielrf12@gmail.com>
Date: Fri, 29 May 2020 18:14:35 -0400
Subject: [PATCH 2/3] libguestfs: mount readonly

---
 diffoscope/comparators/fsimage.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/diffoscope/comparators/fsimage.py b/diffoscope/comparators/fsimage.py
index 9b940327..1c099352 100644
--- a/diffoscope/comparators/fsimage.py
+++ b/diffoscope/comparators/fsimage.py
@@ -62,7 +62,7 @@ class FsImageContainer(Archive):
             return None
         devices = self.g.list_devices()
         try:
-            self.g.mount(devices[0], '/')
+            self.g.mount_options("ro", devices[0], '/')
         except RuntimeError:
             logger.exception("guestfs count not mount image; invalid file?")
             return None
-- 
2.26.1

