diff --git a/diffoscope/comparators/fsimage.py b/diffoscope/comparators/fsimage.py
index 9b940327..1c099352 100644
--- a/diffoscope/comparators/fsimage.py
+++ b/diffoscope/comparators/fsimage.py
@@ -62,7 +62,7 @@ class FsImageContainer(Archive):
             return None
         devices = self.g.list_devices()
         try:
-            self.g.mount(devices[0], '/')
+            self.g.mount_options("ro", devices[0], '/')
         except RuntimeError:
             logger.exception("guestfs count not mount image; invalid file?")
             return None
diff --git a/diffoscope/external_tools.py b/diffoscope/external_tools.py
index 6b97a346..0b758275 100644
--- a/diffoscope/external_tools.py
+++ b/diffoscope/external_tools.py
@@ -220,7 +220,10 @@ EXTERNAL_TOOLS = {
 # May be populated at runtime by remapped names like
 # readelf -> arm-none-eabi-readelf, etc
 # diff -> gdiff, etc
-REMAPPED_TOOL_NAMES = {}
+REMAPPED_TOOL_NAMES = {
+    'readelf': 'aarch64-unknown-linux-gnu-readelf',
+    'objdump': 'aarch64-unknown-linux-gnu-objdump',
+}
 
 # GNU programs whose names differ on some non-GNU systems such as FreeBSD etc
 # AND where the CLI or output of the programs differ from the non-GNU system
