From 008f23a0e3df2fa4ff5afe063cbaa3c12c29ec4f Mon Sep 17 00:00:00 2001
From: cyclopentane <cyclopentane@aidoskyneen.eu>
Date: Thu, 4 Sep 2025 13:19:06 +0200
Subject: [PATCH] Don't run repo during build

---
 build/core/default_installed_modules.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/build/core/default_installed_modules.mk b/build/core/default_installed_modules.mk
index 1f5ba40c..195a89c7 100644
--- a/build/core/default_installed_modules.mk
+++ b/build/core/default_installed_modules.mk
@@ -1,3 +1,2 @@
 # build-manifest.xml
 INSTALLED_BUILD_MANIFEST_XML_TARGET := $(TARGET_OUT_PRODUCT)/etc/build-manifest.xml
-ALL_DEFAULT_INSTALLED_MODULES += $(INSTALLED_BUILD_MANIFEST_XML_TARGET)
-- 
2.50.1

